sawmill-core/src/main/java/io/logz/sawmill/utilities/Grok.java

```diff
@@ -93,10 +93,14 @@ private void addPattern(String patternName, String definition) {
     }
 
     public List<Match> matches(String text) throws InterruptedException {
+        if (Thread.currentThread().isInterrupted()) {
+            throw new InterruptedException();
+        }
```

Comment:
you have to clear the isInterrupted flag once you inside `if`

Thread.interrupted()

Reply:
nice catch!